README.md
[Ruxe Engine](http://ruxe-engine.ru/) - CMS на файлах
=====================================================

О проекте
---------------------------------

Ruxe Engine (далее RE) — это лёгкая, быстрая и бесплатная Open Source система управления сайтом (CMS) на PHP. Она не использует базы данных (такие как, например, MySQL), работает на файлах. Но заострять внимание на этом факте не стоит, ибо это просто способ хранения данных. Для небольших порталов, блогов и сайтов компаний, использование баз данных не оправдывает себя — это просто тяжёлый балласт, замедляющий генерацию страниц.

Суть RE в особом подходе к управлению сайтом. Его можно назвать классическим, продвинутым. Главная идея RE — это гибкость. Нынче мода упрощать до безобразия CMS: установка в один клик, настройка из 5 переключателей и т.п. В итоге, чтобы что-то изменить под себя, приходится или копаться в исходнике, или пытаться это реализовать с помощью редактирования десятков tpl файлов, составляющих шаблон оформления. Порой даже узнать кто присутствует на сайте в данный момент нельзя... Такие CMS предназначены для создания однотипных сайтов за 2 минуты.

RE же предоставляет глубокую настройку вплоть до мельчайших деталей. С его помощью можно сделать непохожий на другие небольшой портал, сайт-визитку, блог и др. С дизайном возиться долго не потребуется — основа шаблонизации лишь один файл. Плюс ещё 6 файлов для тонкой настройки. Для вызова той или иной функции RE используются специальные слова-команды в шаблонах и страницах. В шаблонах не запрещается использовать PHP-код, как и в страницах.

RE работает полностью в UTF-8 кодировке, поддерживает кириллические домены. Весь генерируемый на сайте HTML 4.01 код и RSS-лента проходят проверку на W3.org. Немалое внимание в Ruxe Engine уделяется и безопасности — защита от флуда, премодерация, запрет кэширования страниц, бан посетителей, сессии админ-центра по IP, серьёзная капча (сложная для спам-ботов, но удобная для людей), закрытие определённых страниц от гостей и др.

И напоследок, краткий перечень основных функций RE: шаблонизация, не запрещающая использовать язык PHP; создание и редактирование страниц сайта в админ-центре, которые можно также редактировать и любым другим внешним редактором; ЧПУ; неограниченное число новостных разделов; RSS лента; комментирование в новостях, гостевой книге и отдельных разделах комментариев; регистрация пользователей с аватарами, подписями, активацией по e-mail, восстановлением паролей и прочим; модуль F.A.Q.; каталог файлов; каталог ссылок; ротатор баннеров. Начиная с 0.6.5 версии, появилась поддержка популярного сервиса gravatar.com. Но функционал CMS не ограничивается лишь стандартными модулями — есть и [дополнения](http://ruxe-engine.ru/plugins.html), а новые дополнения можно написать самому. Функционал RE постоянно совершенствуется.


Требования к серверу
---------------------------------
1. Apache 2 с установленным и разрешённым mod_rewrite
2. PHP версии 5.2.17 или новее
3. mbstring должен быть разрешён
4. PHP.ini должен содержать register_globals = Off
5. Установленная библиотека GD 2 для PHP
6. Возможность отправки писем через функцию PHP mail()
7. iconv разрешён (желательно)


Установка
---------------------------------

1. Убедитесь, что конфигурация сервера удовлетворяет требованиям выше.
2. Скачайте стабильный релиз Ruxe Engine по адресу [ruxe-engine.ru/download](http://ruxe-engine.ru/download).
3. Распакуйте скачанный архив на сервер (в корень не обязательно - можно и в подкаталог).
4. Перейдите по http://адрес_сайта/install.php (или http://адрес_сайта/подкаталог/install.php).
5. Следуйте инструкциям.


Применение нового дизайна
---------------------------------

### 1. Файловая структура темы оформления

Тема оформления должна иметь следующую файловую структуру:

    /themes/
        /название_темы/
            commentform.html
            index.php
            list.dat
            message.html
            newsfullrecord.html
            newsrecord.html
            other.php
            users.php
            и другие файлы и каталоги на своё усмотрение

Рассмотрим каждый элемент по отдельности.

 Элемент             | Описание
 ------------------- | --------
 название_темы       | Папка с файлами темы оформления. Каждая папка в каталоге themes - новая тема. Имя папки соответствует её названию. В качестве имён рекомендуется использовать только строчные буквы латинского алфавита и цифры.
 commentform.html    | Здесь находится оформление форм отправки комментариев и личных сообщений. Возьмите его из default темы оформления и вносите изменения, если необходимо.
 index.php           | Шаблон оформления всех страниц сайта.
 newsfullrecord.html | Здесь указывается стиль оформления **полной** записи новостного раздела. Рекомендуется за основу взять шаблон из default темы оформления.
 newsrecord.html     | Здесь находится оформление **краткой** записи новостного раздела. Рекомендуется за основу взять шаблон из default темы оформления. Краткие записи отображаются в списке новостей.
 other.php           | В файле указывается оформление различных модулей движка и стили их записей. С нуля создавать его нельзя - необходимо брать его из default темы оформления и вносить в него нужные изменения.
 users.php           | Содержит в себе оформление всех пользовательских форм, меню и т.п. С нуля создавать его нельзя - необходимо брать его из default темы оформления и вносить в него нужные изменения.

А в list.dat файле прописываются имена пользовательских файлов, находящихся в папке темы, которые будут отображены в правой части раздела "Оформление" Админ-центра, в конце списка, и будут доступны для редактирования встроенным редактором.
Каждый новый файл - на новой строке.
Синтаксис файла следующий:

	имя_файла.расширение|описание_файла

Пример:

	menu.html|Главное меню на сайте

В файлах с расширением .php допускается применять любые языки вёрстки (например, HTML) и веб-программирования, включая серверные (например, PHP). А в файлах с расширением .html - только клиентские языки веб-программирования (например, JavaScript, VisualBasic) и вёрстки. В указанных типах файлов используются HTML-теги и специальные команды движка.


### 2. Пример оформления простейшего шаблона страницы в файле index.php

```html
<!-- Тип документа и синтаксиса языка разметки; RuxeEngine адаптирован под HTMLv4.01 Transitional -->
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
        <!-- Заголовок страницы -->
        <title><?php here_title(); ?></title>
        <!-- ОБЯЗАТЕЛЬНАЯ команда, которая выводит meta-тег с переадресацией в промежуточных страницах; при отключенных промежуточных страницах оставляет на месте себя только перенос строки -->
        <?php here_metaredirect(); ?>
        <!-- Параметры RSS-ленты -->
        <link rel="alternate" type="application/rss+xml" title="RSS-лента" href="<?php here_urlsite(); ?>/rss">
        <!-- Иконка сайта -->
        <link rel="SHORTCUT ICON" href="<?php here_urlsite(); ?>/favicon.ico">
        <!--Тип содержимого и кодировка страницы-->
        <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
        <!--Описание страницы-->
        <meta name="description" content="<?php here_description(); ?>">
        <!--Ключевые слова страницы-->
        <meta name="keywords" content="<?php here_keywords(); ?>">
</head>
<body>
        <!-- Здесь может быть заголовок, слоган сайта, главное меню, сайдбар и др.-->
        <h1>МОЙСАЙТ</h1>
        <!-- Команда для вывода контента той или иной страницы -->
        <?php here_pagecontent(); ?>
        <!-- Команда для вывода текста "Powered by Ruxe Engine" с указанием ссылки на официальный сайт (команда необязательная, но мы надеемся, что вы её будете использовать); местоположение данной команды может быть в любом месте страницы в пределах тегов <body></body>-->
        {CREDITS}
</body>
</html>
```


Что такое «команды» в Ruxe Engine
---------------------------------

Команда представляет собой определённую фразу, интерпретировав которую, движок выполняет ту или иную функцию, выдавая результат этой функции, либо подставляет значение вместо указанной переменной.

Команды делятся на те, которые употребляются только на страницах и в главном шаблоне – index.php (Имеется ввиду index.php, находящийся в /themes/**название_темы**/. Здесь и далее не путать с index.php, расположенным в корне сайта), и те, которые присутствуют в других шаблонах темы оформления.

**Пример 1**. На страничке сайта нужно вывести форму авторизации на сайте. Для этого в основном шаблоне страницы (админ-центр, раздел «Оформление») в нужном месте вставляем команду **<?php here_login(); ?>** и сохраняем изменения.

```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
        <head>
                <title>Заголовок</title>
        </head>
<body>
<h1>Мой сайт</h1><br>
<h2>Форма авторизации:</h2>
<?php here_login(); ?>
</body>
</html>
```

При просмотре сайта через браузер, в том месте, где была вставлена команда, будет находиться форма авторизации. Аналогичным образом можно вставить форму Обратной связи, список категорий новостных разделов, статистику посещения и др.

**Пример 2**. При публикации записи система автоматически подставляет некоторые данные вместо переменных, указанных в шаблоне краткой новости: заголовок статьи ({TITLE}), категорию ({CATEGORY}) со ссылкой на адрес категории ({CATURL}), автора ({AUTHOR}) со ссылкой на его профиль, дату публикации ({DATE}), счётчик количества просмотров ({VIEWS}) и количество комментариев ({COMMENTS}) со ссылкой на них ({COMURL}). Отредактировав файл шаблона краткой новости (админ-центр, раздел «Оформление»), можно изменить порядок их вывода, либо совсем убрать некоторые из них.

Существуют также обязательные команды: если их не будет в основном шаблоне сайта, то движок, либо некоторые его функции работать не будут. К ним относятся **<?php here_metaredirect(); ?>** (выводит meta-тег с переадресацией в промежуточных страницах, размещать её нужно сразу после закрывающего тега ) и **{CREDITS}** (выводит текст "Powered by Ruxe Engine" со ссылкой на официальный сайт движка, она может быть расположена в любом месте страницы в пределах тегов, начиная с версии 1.72 команда не является обязательной, но мы надеемся, что вы будете её использовать).

Список стандартных команд движка может быть расширен благодаря установленным дополнениям.


Список большинства «команд»
---------------------------------

### Команды страниц, а также шаблона index.php

Употребляются в исходном коде страниц, использующих Ruxe Engine, и в шаблоне index.php для активации тех или иных функций при последующем просмотре страниц в браузере.


	<?php here_pagecontent(); ?>


history.txt
1.5
	В одиночных разделах комментариев добавлена возможность отключения комментирования без скрытия уже 
		существующих комментариев;
	В метках новостей заменяются + на p, а # на Sharp;
	Исправлена ошибка, возникающая при указании в cookie неустановленного языка;
	Забаненный пользователь больше не может восстановить себе пароль, что приводило к разблокировке;
	Регистрация пользователей теперь осуществляется по адресу http://site/newuser или http://site/?action=newuser
		(против ботов). Скрипт обновления пытается обновить .htaccess, однако если .htaccess не обновился, то зайдите
		в админ-центр - Настройки - Сохранить настройки;
	Устранено недоразумение, из-за которого mbstring не использовался, когда мог, для повышения производительности;
	Обновлён перевод;
	Также устранены мелкие огрехи в административном центре, приводящие к возникновению Notice'ов;
	Обновлена система обнаружения поисковых роботов;
	Обновлено лицензионное соглашение;
	Увеличена ширина столбцов, отображающих даты;
	Обновлены ссылки на официальный сайт в соответствии с изменениями, оговоренными в начале новости
		http://ruxesoft.net/articles/sayt-ruxe-engine-budet-perenesen.html

1.4.1
	Небольшое исправление в админ-центре

1.4
	1. Появилась возможность выбирать посетителям сайта язык. Для этого с помощью JS или PHP скриптов администратор должен
		позволить посетителям установить куку под именем lang. В качестве значения lang принимает только имена подкаталогов в 
		includes/languages/.
	2. На главной странице админ-центра теперь можно работать с последними комментариями (изменять, удалять и т.п.)
	3. Новая команда - <? here_top_news('ID'); ?>, выводящая список самых популярных новостей в разделе по идентификатору ID
		Максимальное количество выводимых ссылок указывается в Настройках админ-центра
		Оформление в other.php $ocms[30]
	4. Исправлено: существовала низкая вероятность, что добавленный комментарий и комментарий в списке Последних комментариев
		будут иметь разные даты, что вызывало невозможность управлять этим комментарием из Последних комментариев
	5. Обновлено оформление админ-центра, поэтому Даниилу необходимо обновить rpanel.css тёмного стиля админ-центра
		в соответствии с новой версией
	6. Обновление системы шаблонизации до версии 0.2
	7. При создании и редактировании новостей в список уже использованных меток имеет правильную сортировку независимо от
		регистра букв, но только в том случае, если установлено дополнение iconv на сервере
	8. Исправлено: категория новостей, которая используется только в неопубликованных новостях, отображалась в here_list_category и др.
		местах
	9. Внесены изменения в инициализацию дополнений, что позволит существенно увеличить возможности
	10. Теперь можно регистрировать пользователей из админ-центра
	11. При просмотре профиля пользователя на сайте - дата регистрации отображается с учётом выбранного часового пояса
	12. Некоторые сообщения об ошибках приведены в приятный вид
	13. Исправлено: если имеются два или более одинаковых комментариев, отправленных в одно и то же время, то при удалении
		любого из этого комментария - в Последних комментариях удаляются упоминания обо всех похожих комментариях
	14. Исправлено: если в шаблоне newsrecord.html или newsfullrecord.html не использовать определения меток, то появляются
		ошибки
	15. Улучшена система распознавания ошибок в шаблонизации
	16. При создании и редактировании новостей, если создавать новость-ссылку, то с "Разрешить комментарии" автоматически снимается
		галочка
	
	
1.3.6
	Исправлено: если навигация не обратная, то ссылки "новее" и "ранее" перепутаны местами
1.3.5
	1. Новости теперь можно переносить из одного новостного раздела в другой
	2. Новое оформление списка новостей в админ-центре
	3. Исправлено: Undefined offset: 1 в файле \rpanel\index.php на строке 763
	4. Подправлено оформление блоков разделов новостных и комментариев админ-центра в браузере опера
	5. Команда <? here_downloads(); ?> в следующих версиях будет отменена. Об этом будет сообщаться ещё несколько
		версий при просмотре вывода результата работы команды администратором.
	6. Обновление темы оформления сайтов
	7. В Каталоге ссылок, при включённой опции запрета индексации ссылок поисковыми системами, вместо <noindex></noindex> теперь
		используется валидный rel="nofollow"
	8. Введена система распознавания проблем с темами оформления, оформленными не по правилам шаблонизации актуальной версии
	9. В разделе "Счётчики загрузок" теперь отображается количество загрузок для каждой записи
	10. Обновление интерфейса изменения прав пользователей
	11. При использовании визуального редактора TinyMCE опция "Заменять обычные кавычки (") на « »" более недоступна
	12. Появилась возможность отключить обратную навигацию в Настройках админ-центра
	
1.3.2
	1. Подправлено оформление админ-центра
	2. Исправлено: в описаниях счётчиков загрузок удалялись пробелы
	3. Исправлено: в логе бана неправильно сохранялась и отображалась дата
	4. В разделе Обслуживание админ-центра список последних бан записей имеет правильную дату
	5. Обновлена тема оформления админ-центра таким образом, чтобы Даниил мог изменить стиль под тёмный
	
1.3.1
	Исправлено: навигация не отображалась в админ-центре
1.3
	1. Шаблоны index.php, other.php и users.php больше не подключаются в админ-центре, что предоставляет возможность

rpanel/users.php
<?php

/*
 * Ruxe Engine - CMS на файлах
 * http://ruxe-engine.ru
 *
 * Лицензия:
 * Это произведение доступно по Open Source лицензии
 * Creative Commons «Attribution-ShareAlike» («Атрибуция — На тех же
 * условиях») 4.0 Всемирная (CC BY-SA 4.0).
 *
 * Разработчики:
 * Ахрамеев Денис Викторович (http://ruxesoft.net) Автор, программирование
 * Игорь Dr1D - Дизайн
 * Олег Прохоров (http://www.tanatos-life.ru) - Контроль качества, документация
 *
 */

	include("check2.php");

	$echooptions 	= "";
	$menu 		= "";
	$mess 		= "";
	$users 		= file("../conf/users/users.dat");
	$start 		= '<h2>Пользователи и администраторы</h2>
				<font class="desc">В данном разделе Вы можете удалить, забанить, изменить профиль пользователей, установить статус (например, администратор или суперпользователь), обнулить счётчики сообщений, добавить новые поля профиля и страницы, доступные для просмотра только администраторам и суперпользователям</font><br><br>
			';
	$echooptions 	= "<center><table class=\"optionstable\" border=0 cellpadding=1 cellspacing=0>
				<tr class=\"titletable\"><td>ИМЯ ПОЛЬЗОВАТЕЛЯ</td><td width=110>СТАТУС (<a target=\"_blank\" title=\"Помощь\" href=\"http://ruxe-engine.ru/documentation/roles.html\">?</a>)</td><td width=85>СООБЩЕНИЙ</td><td width=150>E-MAIL</td><td width=100>IP</td><td width=270>ДЕЙСТВИЕ</td></tr>
			";
	if (isset($_GET["poles"]))
		$mess 	= "(Сохранено)";

	include("../conf/users/config.dat");
	
	if (isset($_GET['go'])) {
		$go 	= $_GET['go'];
		$user 	= $_GET['user'];
		if ($go == 'new') {
			$menu .= '<form name="newuser" action="saver.php?saverdo=newuser" method="POST">
					<center><table class="optionstable2" border=0 cellpadding=1 cellspacing=0>
						<tr class="titletable"><td colspan=2>СОЗДАНИЕ НОВОГО ПОЛЬЗОВАТЕЛЯ</td></tr>
						<tr><td>Логин:</td><td><input type="text" size=30 name="login"></td></tr>
						<tr><td>Пароль:</td><td><input type="password" size=30 name="password"></td></tr>
						<tr><td>Повторите пароль:</td><td><input type="password" size=30 name="spassword"></td></tr>
						<tr><td>Email:</td><td><input type="text" size=30 name="mail"></td></tr>
						<tr><td>Повторите email:</td><td><input type="text" size=30 name="smail"></td></tr>
						<tr class="sub"><td colspan=2><input type="submit" name="submit" value="Создать"></td></tr>
					</table></center>
					</form><br>';
		}
		
		if ($go=="edit") {
			for ($i=0; $i<count($users); $i++) {
				$p = explode("|",$users[$i]);
				if ($user == $p[0])
					$pos = $i;
			};
			$p 	= explode("|",$users[$pos]);
			$menu 	.= '<form name="edituser" action="saver.php?saverdo=edituser" method="POST">
					<center><table class="optionstable" border=0 cellpadding=1 cellspacing=0>
					<tr class="titletable"><td colspan=2>РЕДАКТИРОВАНИЕ ПРОФИЛЯ</td></tr>
					<tr><td>Логин:</td><td width=345><input type="text" size=52 name="login" value="'.$p[18].'"></td></tr>
					<tr><td><input type="hidden" name="lastlogin" value="'.$p[18].'">Пароль <font class="desc">(оставьте пустым, если не хотите менять)</font>:</td><td><input type="password" name="password" size=52></td></tr>
					<tr><td>E-mail:</td><td><input type="text" name="mail" size=52 value="'.$p[2].'"></td></tr>
					<tr><td>О себе:</td><td><textarea name="about" cols=40 rows=3>'.str_replace("<br>","\n",$p[3]).'</textarea></td></tr>
					<tr><td>Подпись:</td><td><textarea name="ps" cols=40 rows=3>'.str_replace("<br>","\n",$p[17]).'</textarea></td></tr>
					<tr><td>Звание (статус) <a title="Помощь" href="http://ruxe-engine.ru/documentation/roles.html" target="_blank">?</a>:</td><td>
				';
			if ($pos!=0) {
				$menu .='<select name="status" id="statuser" onChange="diuc()">';
        
				switch ($p[4]) {
					case "admin":
						$menu.= '<option value="admin" selected>'.$lcms['user_status_admin'].'
							<option value="superuser">'.$lcms['user_status_superuser'].'
							<option value="user">'.$lcms['user_status_user'].'
							<option value="baned">'.$lcms['user_status_baned'].'
							<option value="active">'.$lcms['user_status_noactive'].'
							<option value="editor">'.$lcms['user_status_editor'].'
							<option value="moderator">'.$lcms['user_status_moderator'].'
						';
						break;
					case "superuser":
						$menu.= '<option value="admin">'.$lcms['user_status_admin'].'
							<option value="superuser" selected>'.$lcms['user_status_superuser'].'
							<option value="user">'.$lcms['user_status_user'].'
							<option value="baned">'.$lcms['user_status_baned'].'
							<option value="active">'.$lcms['user_status_noactive'].'
							<option value="editor">'.$lcms['user_status_editor'].'
							<option value="moderator">'.$lcms['user_status_moderator'].'
						';
						break;
					case "user":
						$menu.= '<option value="admin">'.$lcms['user_status_admin'].'
							<option value="superuser">'.$lcms['user_status_superuser'].'
							<option value="user" selected>'.$lcms['user_status_user'].'
							<option value="baned">'.$lcms['user_status_baned'].'
							<option value="active">'.$lcms['user_status_noactive'].'
							<option value="editor">'.$lcms['user_status_editor'].'
							<option value="moderator">'.$lcms['user_status_moderator'].'
						';
						break;
					case "baned":
						$menu.= '<option value="admin">'.$lcms['user_status_admin'].'
							<option value="superuser">'.$lcms['user_status_superuser'].'
							<option value="user">'.$lcms['user_status_user'].'
							<option value="baned" selected>'.$lcms['user_status_baned'].'
							<option value="active">'.$lcms['user_status_noactive'].'
							<option value="editor">'.$lcms['user_status_editor'].'
							<option value="moderator">'.$lcms['user_status_moderator'].'
						';
						break;
					case "active":
						$menu.= '<option value="admin">'.$lcms['user_status_admin'].'
							<option value="superuser">'.$lcms['user_status_superuser'].'
							<option value="user">'.$lcms['user_status_user'].'
							<option value="baned">'.$lcms['user_status_baned'].'
							<option value="active" selected>'.$lcms['user_status_noactive'].'
							<option value="editor">'.$lcms['user_status_editor'].'
							<option value="moderator">'.$lcms['user_status_moderator'].'
						';
						break;
					case "editor":
						$menu.= '<option value="admin">'.$lcms['user_status_admin'].'
							<option value="superuser">'.$lcms['user_status_superuser'].'
							<option value="user">'.$lcms['user_status_user'].'
							<option value="baned">'.$lcms['user_status_baned'].'
							<option value="active">'.$lcms['user_status_noactive'].'
							<option value="editor" selected>'.$lcms['user_status_editor'].'
							<option value="moderator">'.$lcms['user_status_moderator'].'
						';
						break;
					case "moderator":
						$menu.= '<option value="admin">'.$lcms['user_status_admin'].'
							<option value="superuser">'.$lcms['user_status_superuser'].'
							<option value="user">'.$lcms['user_status_user'].'
							<option value="baned">'.$lcms['user_status_baned'].'
							<option value="active">'.$lcms['user_status_noactive'].'
							<option value="editor">'.$lcms['user_status_editor'].'
							<option value="moderator" selected>'.$lcms['user_status_moderator'].'
						';
						break;
				};
        
				$display= 'none';
				if ($p[4] == 'baned') 
					$display = 'inline';
				if (($p[4] == 'editor') or ($p[4]=='moderator')) 
					$display_editor = 'inline';
				else
					$display_editor = 'none';
				if ($p[4]=='moderator')
					$display_moderator	=	'inline';
				else
					$display_moderator	=	'none';
        
				$menu	.= '</select> 
						<div style="display:'.$display.'; margin-top:2px; margin-bottom: 2px;" id="whyban"><br>Причина:<br><input name="whybaned" type="text" id="wb1" style="margin-top:3px;" size=52 value="'.$p[19].'"></div>
					';
				//Редактор
				$menu.='
					<div style="display:'.$display_editor.'; margin-top:2px; margin-bottom: 2px;" id="display_editor">
						<br>Новостные разделы: <br>
        		';
        $bfg		=	file($cms_root.'/conf/bfg.dat');
        $rules		=	$GlobalUsers->getrules($p[18]);
        $bfg_types	=	explode(',',$rules['bfg_types']);
        foreach ($bfg as $bf)
        {
        	$b 		=	explode('|',$bf);
        	$checked 	= 	'';
        	foreach ($bfg_types as $z)
        	{
        		if ($z==$b[0])
        		{
        			$checked = ' checked';
        			break;
        		};
        	};
        	$menu .= '<input type="checkbox" name="editor_'.$b[0].'" value="'.$b[0].'"'.$checked.'> '.$b[1].'<br>';
        };
        if ($rules['bfg_public']==true)
        	$checked = ' checked';
        else
        	$checked = '';
        if ($rules['bfg_delete']==true)
        	$bfg_delete	=	' checked';
        else
        	$bfg_delete	=	'';
        $menu.='
        	<br><input type="checkbox" name="bfg_public" value="yes"'.$checked.'> Разрешить опубликовывать новости<br>
		<input type="checkbox" name="bfg_delete" value="yes"'.$bfg_delete.'> Разрешить удалять новости
        	</div>
        ';
        
        //
        //Модератор
        if ($rules['comments_edit']==true)
        	$comments_edit	=	' checked';
        else
        	$comments_edit	=	'';
        if ($rules['plugins_use']==true)
        	$plugins_use	=	' checked';
        else
        	$plugins_use	=	'';
        if ($rules['banip']==true)
        	$banip		=	' checked';
        else
        	$banip		=	'';
	//<br><input type="checkbox" name="banip" value="yes"'.$banip.'> Блокировка посетителей по IP (<font color="red"><b>В разработке</b></font>)
        $menu.='<div style="display:'.$display_moderator.'; margin-top:2px; margin-bottom:2px;" id="display_moderator">
        	<br><input type="checkbox" name="comments_edit" value="yes"'.$comments_edit.'> Управление комментариями
        	<br><input type="checkbox" name="plugins_use" value="yes"'.$plugins_use.'> Использование дополнений
        	<input type="hidden" name="banip" value="">
        </div>';
        //
        
        }
        else
        {
            $menu.='<b>Главный администратор</b>';
        }
	if (
		($p[20]!='Не известно')
		and
		($p[20]!='Начало эпохи Unix')
		)
		$p[20]	=	$Filtr->fulldate(
					str_replace('2011','11',$p[20])
					);
        $menu.='
        </td></tr>
	<tr><td>IP при регистрации:</td><td>'.$Filtr->ipclick($p[12]).'</td></tr>
        <tr><td>Дата регистрации:</td><td>'.str_replace('Не известно','Неизвестно',$p[20]).'</td></tr>
        ';
        for ($i = 1; $i<=7; $i++)
        {
            if ($pole[$i]==true)
            {
                $menu.='<tr><td>'.$polecaption[$i].':</td><td><input type="text" name="pole'.$i.'" value="'.$p[5+$i-1].'" size=52></td></tr>
                ';
            }
            else
            {
                $menu.='<input type="hidden" name="pole'.$i.'" value="'.$p[5+$i-1].'">
                ';
            };
        };         
	$ava	=	$cms_site.'/avatars/'.$Filtr->clear($p[16]);
	if ($p[16]=='noavatar.png')
	{
		if ($cms_gravatars==1)
			$ava	=	$GlobalImaging -> get_gravatar($p[2],$cms_upload_width,$cms_gravatars_im,'g',false);
	};	
        $menu.='<tr><td>Сообщений:</td><td><input type="text" size=52 name="messages" value="'.$p[14].'"></td></tr>
        <tr><td>Аватар:</td><td>
        <input type="hidden" name="avatar" id="av3" value="'.$p[16].'">
        <input type="hidden" name="oldavatar" value="'.$p[16].'">
        <img id="av" alt="" src="'.$ava.'" border=0> <input id="av2" type="button" onClick="document.getElementById(\'av\').style.display=\'none\'; document.getElementById(\'av2\').style.display=\'none\'; document.getElementById(\'av3\').value=\'noavatar.png\';" value="Удалить"></td></tr>
        <tr class="sub"><td colspan=2><input type="submit" name="submit" value="Сохранить"></td></tr>
        </table>
        </center>
        </form><br>';
     };
  };
	for ($i=0; $i<count($users);$i++) {
		$p 	= explode("|",$users[$i]);
		$name 	= $p[18];
		switch ($p[4]) {
			case "admin":
				$status	= $lcms['user_status_admin'];
				$name 	= "<font color=\"red\"><b>".$name."</b></font>";
				break;
			case "moderator":
				$status = $lcms['user_status_moderator'];
				$name 	= "<font color=\"green\"><b>".$name."</b></font>";
				break;
			case "editor":
				$status = $lcms['user_status_editor'];
				$name 	= "<font color=\"green\"><b>".$name."</b></font>";
				break;
			case "superuser":
				$status = $lcms['user_status_superuser'];
				$name 	= "<font color=\"green\"><b>".$name."</b></font>";
				break;
			case "user":
				$status = $lcms['user_status_user'];
				break;
			case "baned":
				$status = $lcms['user_status_baned'];
				$name = "<font color=\"grey\"><b>".$name."</b></font>";
				break;
			case "active":
				$status = $lcms['user_status_noactive'];
				$name = "<font color=\"grey\"><b>".$name."</b></font>";
				break;
		}
		if (
			($p[20]!='Не известно')
			and
			($p[20]!='Начало эпохи Unix')
		)
			$p[20]	=	$Filtr->fulldate(str_replace('2011','11',$p[20]));
		$echooptions.="<tr><td>";
		$echooptions	.= "<a href=\"".$Navigation->furl('viewprofile',$p[18])."\" onClick=\"window.open('".$Navigation->furl('viewprofile',$p[18])."'); return false;\">".$name."</a><font class=\"desc\"><br>Зарегистрирован: ".str_replace('Не известно','Неизвестно',$p[20])."</font></td><td>";
		if ($i==count($users)-1) {
			$echooptions.='
					<div id="newuser" style="position: absolute; display: none;">
					<form name="newuser" action="saver.php?saverdo=newuser" method="POST">
					<table class="optionstable2" border=0 cellpadding=1 cellspacing=0>
						<tr class="titleblue"><td colspan=2>СОЗДАНИЕ НОВОГО ПОЛЬЗОВАТЕЛЯ</td></tr>
						<tr class="bluetd"><td>Логин:</td><td><input type="text" size=30 name="login"></td></tr>
						<tr class="bluetd"><td>Пароль:</td><td><input type="password" size=30 name="password"></td></tr>
						<tr class="bluetd"><td>Повторите пароль:</td><td><input type="password" size=30 name="spassword"></td></tr>
						<tr class="bluetd"><td>Email:</td><td><input type="text" size=30 name="mail"></td></tr>
						<tr class="bluetd"><td>Повторите email:</td><td><input type="text" size=30 name="smail"></td></tr>
						<tr class="bluetd"><td colspan=2><input type="submit" name="submit" value="Создать"></td></tr>
					</table>
					</form>
					</div>';
		}
		$echooptions	.= $status."</td><td>".$p[14]."</td><td>".$p[2]."</td><td>".$Filtr->ipclick($p[12])."</td>
					<td>
					<a href=\"?action=users&amp;go=edit&amp;user=".$p[0]."&amp;rand=".rand(0,99999)."\">Изменить</a>";
		if ($i!=0) {
			if ($p[4]=="baned")
				$echooptions.= ' <a href="saver.php?saverdo=banuser&amp;user='.$p[0].'&amp;donotban=true">Разблокировать</a>';
			else
				$echooptions.= ' <a href="saver.php?saverdo=banuser&amp;user='.$p[0].'">Заблокировать</a>';
			$echooptions.=" <a href=\"#\" onClick=\"if (checkhead()) location.href='saver.php?saverdo=deluser&amp;user=".$i."';\">Удалить</a>";
		};
		$echooptions.="</td></tr>";
	}; 

  $echooptions .='
  <tr class="sub"><td colspan=6>
	<input type="button" onClick="document.getElementById(\'newuser\').style.display=\'inline\';" value="Создать нового пользователя">
	</td></tr>
  </table><br>
  <form name="poles" action="saver.php?saverdo=savepoles" method="POST">
   <table class="optionstable" border=0 cellpadding=1 cellspacing=0>
    <tr class="titletable"><td width="50%">ДОПОЛНИТЕЛЬНЫЕ ПОЛЯ В ПРОФИЛЕ</td><td width="50%">ОГРАНИЧЕННЫЙ ДОСТУП</td></tr>
    <tr><td valign="top" align="center"><div align="left"><input type="button" onClick="addpole();" value="Добавить поле"></div><br>
    ';
    for ($i=1; $i<=7; $i++)
    {
      $echooptions.='<input style="display: inline" type="text" id="pole'.$i.'" name="pole'.$i.'" value="'.$polecaption[$i].'" size=50>
      <input type="button" style="display: inline" onClick="delpole(\'pole'.$i.'\');" id="pole'.$i.'b" value="Удалить"><div style="display:block;" id="pole'.$i.'c"><br></div>
';
    };
  $pagesconfig = file("../conf/pages/config");
  $echooptions .='
   </td><td valign="top" align="center">
    <div align="left">Адреса страниц сайта, просмотр которых разрешён только администраторам, модераторам, редакторам и суперпользователям
    <br>
    <select style="margin-top:10px;" id="newpage">
    ';
    $c = 0;
    foreach ($pagesconfig as $pageline)
    {
        $p = explode("|",$pageline);
	if ($p[0]=='index') 
	{
		$p[0]='index';
		$shw	=	'index.php';
	}
	else
		$shw	=	$p[0];
        ($c==0) ? $echooptions .= '<option value="/'.$p[0].'" selected>'.$shw : $echooptions .='<option value="/'.$p[0].'">'.$shw;
        $c++;
    };
    $echooptions.='
    </select> <input type="button" onClick="addpage();" value="Добавить страницу"></div><br>
   
   ';
   $hidepages = file("../conf/users/hidepages.dat");
   $pagecaption = explode("|",$hidepages[0]);
    for ($i=0; $i<=9; $i++)
    {
      
      $echooptions.='<input style="display: inline" readonly type="text" id="page'.$i.'" name="page'.$i.'" value="'.$pagecaption[$i].'" size=50>
      <input type="button" style="display: inline" onClick="delpage(\'page'.$i.'\');" id="page'.$i.'b" value="Удалить"><div style="display:block;" id="page'.$i.'c"><br></div>
      ';
      
    };
   $echooptions .='</td></tr>
   <tr class="sub"><td colspan=2><input type="submit" name="submit" value="Сохранить все изменения"> '.$mess.'</td></tr>  
   </table>
  </form>
  </center>
  ';
  $ar = array("{MENU}","{OPTIONS}");
  $br = array("",$start.$menu.$echooptions);
  echo $GlobalTemplate->template($ar,$br,"./theme/admincenteroptions.tpl");

